src/platform/datapath_raw_xdp.c

```diff
@@ -1077,88 +1077,93 @@ CxPlatDpRawInitialize(
         Xdp,
         Xdp->WorkerCount);
 
-    PIP_ADAPTER_ADDRESSES Adapters = NULL;
-    ULONG Error;
-    ULONG AdaptersBufferSize = 15000; // 15 KB buffer for GAA to start with.
-    ULONG Iterations = 0;
-    ULONG flags = // skip info that we don't need.
-        GAA_FLAG_INCLUDE_PREFIX |
-        GAA_FLAG_SKIP_UNICAST |
-        GAA_FLAG_SKIP_ANYCAST |
-        GAA_FLAG_SKIP_MULTICAST |
-        GAA_FLAG_SKIP_DNS_SERVER |
-        GAA_FLAG_SKIP_DNS_INFO;
-
-    do {
-        Adapters = (IP_ADAPTER_ADDRESSES*)CxPlatAlloc(AdaptersBufferSize, ADAPTER_TAG);
-        if (Adapters == NULL) {
+    PMIB_IF_TABLE2 pIfTable;
+    if (GetIfTable2(&pIfTable) != NO_ERROR) {
+        Status = QUIC_STATUS_INTERNAL_ERROR;
+        goto Error;
+    }
+
+    struct IfMatching {
+        MIB_IF_ROW2* IfRow;
+        MIB_IF_ROW2* ActualIfRow; // for VF
+    };
+
+    struct IfMatching IfMatching[16] = {0}; // TODO: linked list
```

Comment:
So, instead of first building a list of objects, and then going and allocating matching XDP objects for each, why not do it all together? For instance, keep the old loop we had, where we create an XDP object for each IP address/interface; but then inside that loop, go try to find a "matching" interface for the "actual" interface index. If found, you update the actual index.